src/contribution/contribution.service.ts

```diff
@@ -1,17 +1,22 @@
-import { Injectable } from "@nestjs/common";
+import { Injectable, NotFoundException } from "@nestjs/common";
 import { Repository } from "typeorm";
 import { InjectRepository } from "@nestjs/typeorm";
+import { GetPrevDateISOString } from "../common/util/datetimes";
 import { PageDto } from "../common/dtos/page.dto";
 import { OrderDirectionEnum } from "../common/constants/order-direction.constant";
 import { PageMetaDto } from "../common/dtos/page-meta.dto";
+import { DbUser } from "../user/user.entity";
 import { DbContribution } from "./contribution.entity";
 import { ContributionOrderFieldsEnum, ContributionPageOptionsDto } from "./dtos/contribution-page-options.dto";
+import { DbRepoLoginContributions } from "./entities/repo-user-contributions.entity";
 
 @Injectable()
 export class ContributionService {
   constructor(
     @InjectRepository(DbContribution, "ApiConnection")
-    private contributionRepository: Repository<DbContribution>
+    private contributionRepository: Repository<DbContribution>,
+    @InjectRepository(DbUser, "ApiConnection")
+    private usersRepository: Repository<DbUser>
```

Comment:
nit (naming): I'm understanding this as a user's repo?

```suggestion
    private userRepository: Repository<DbUser>
```

Reply:
Good catch! 🙈